idaes/generic_models/unit_models/column_models/column.py

```diff
@@ -0,0 +1,1144 @@
+
+        self.phi = Expression(
+            self.flowsheet().time,
+            self.vapor_phase.length_domain,
+            solute_comp_list,
+            rule=rule_phi,
+            doc='Equilibruim partial pressure intermediate term for solute')
+
+        # Equilibruim partial pressure of diffusing components at interface
+        @self.Constraint(self.flowsheet().time,
+                          self.vapor_phase.length_domain,
+                          equilibrium_comp,
+                          doc='''Equilibruim partial pressure of diffusing
+                                components at interface''')
+        def pressure_at_interface(blk, t, x, j):
+            if x == self.vapor_phase.length_domain.first():
+                return blk.pressure_equil[t, x, j] == 0.0
+            else:
+                zb = self.vapor_phase.length_domain.at(self.zi[x].value - 1)
+                lprops = blk.liquid_phase.properties[t, zb]
+                henrycomp = lprops.params.get_component(j).config.henry_component
+                if henrycomp is not None and "Liq" in henrycomp:
+                    return blk.pressure_equil[t, x, j] == (
+                        (blk.vapor_phase.properties[t, x].mole_frac_comp[j] *
+                          blk.vapor_phase.properties[
+                              t, x].pressure + blk.phi[t, x, j] *
+                          lprops.conc_mol_phase_comp_true['Liq',j]) /
+                        (1 + blk.phi[t, x, j] /
+                          blk.liquid_phase.properties[t, zb].henry['Liq',j]))
+                else:
+                    return blk.pressure_equil[t, x, j] == (
+                        lprops.vol_mol_phase['Liq'] *
+                        lprops.conc_mol_phase_comp_true['Liq',j] *
+                        lprops.pressure_sat_comp[j])
+
+        # Mass transfer of  diffusing components in vapor phase
+        def rule_mass_transfer(blk, t, x, j):
+            if x == self.vapor_phase.length_domain.first():
+                return blk.interphase_mass_transfer[t, x, j] == 0.0
+            elif j in equilibrium_comp:
+                return blk.interphase_mass_transfer[t, x, j] == (
+                    blk.k_v[t, x, j] *
+                    blk.area_interfacial[t, x] * blk.area_column *
+                    (blk.vapor_phase.properties[t, x].mole_frac_comp[j] *
+                     blk.vapor_phase.properties[t, x].pressure -
+                     blk.pressure_equil[t, x, j]))
+            else:
+                return blk.interphase_mass_transfer[t, x, j] == 0.0
+
+        self.mass_transfer_vapor = Constraint(self.flowsheet().time,
+                                        self.vapor_phase.length_domain,
+                                        liq_comp,
+                                        rule=rule_mass_transfer,
+                                        doc="mass transfer in vapor phase")
+
+        # Liquid phase mass transfer handle
+        @self.Constraint(self.flowsheet().time,
```

Comment:
This all looks more complicated than necessary to me right now (I could be wrong). What I was thinking of was:

1. A constraint across all species that says `liquid_phase.mass_transfer[t, x, p, j] == vapor_phase.mass_transfer[t, x, p, j]` for all equilibrium species, `liquid_phase.mass_transfer[t, x, p, j] == 0` for liquid only species and `vapor_phase.mass_transfer[t, x, p, j] == 0` for all vapour only components.
2. A constraint across all equilibrium components that says `liquid_phase.mass_transfer[t, x, p, j] == K*A*A_inter*(y*P - P_equil)`.
3. Some constraint that defines `P_equil` which for the generic model is probably `P_equil[t, x, p, j] == liquid_phase.properties[t, x].fugactiy_phase_comp[p, j]` (but we would need to modify this for more detailed models.
4. `K` should either be a fixed value, or there should be a simple constraint that describes the actual `K` in terms of enhancement factor and `K_prime` measured by experiments.